src/backer/queueing.py

```diff
@@ -0,0 +1,44 @@
+from hio import doing
+from keri.db import subing
+from keri.core import serdering, eventing
+from ... import cardaning
+
+
+class Queueing(doing.Doer):
+
+    def __init__(self, name='backer', hab=None):
+        self.name = name
+        self.hab = hab
+        # TODO: pending_kel should change to array
+        self.pending_kel = None
+        self.ledger = cardaning.Cardano(name=name, hab=hab)
+        # sub-dbs
+        self.keldb_queued = subing.SerderSuber(db=hab.db, subkey="kel_queued")
+        self.keldb_published = subing.SerderSuber(db=hab.db, subkey="kel_published")
+        self.tock = 10  # the job should run daily.
+
+    def publish(self):
+        """
+        doer call run
+            get from queued
+                do publishEvent
+            deleted from queued
+            create in published
+        """
+        for (pre, said), serder in self.keldb_queued.getItemIter():
+            event = eventing.loadEvent(self.hab.db, pre, serder.saidb)
+            self.ledger.publishEvent(event=event)
```

Comment:
hmm, so ledger has `pending_kel` and periodically it pushes that on-chain. It should be updated to accept a batch of KELs and write them on-chain immediately. So the tock is only in this file.

And that way we only call `rem` and `pin` on the DB after we know it successfully wrote something on-chain

Reply:
updated
pls ref the latest commit: a54d02f52ce5fd090055598ddc584c4a25768e54

Reply:
I think you are misunderstanding: when you publish an event from queuing.py - it calls `self.ledger.publishEvent(event=event)` in cardaning.py.

`ledger.publishEvent` overwrites `pending_kel` and after 90 seconds pushes on-chain.

We should update this to push ALL of the queued items from queuing.py in one go, into one tx. And create the tx immediately instead of having this 90 second wait. We already have queueing.py on a tock setting - we don't need 2 waits.

Reply:
In another story, we will handle the case where there are too many queued items to fit in a transaction but that's later

Reply:
checking

Reply:
updated 
ref here: 9e2a6767f6b6e9be924216aa0f9a9954812024d1